var/spack/repos/builtin/packages/curl/package.py

```diff
@@ -49,21 +49,37 @@ class Curl(AutotoolsPackage):
     version('7.43.0', '11bddbb452a8b766b932f859aaeeed39')
     version('7.42.1', '296945012ce647b94083ed427c1877a8')
 
-    variant('nghttp2', default=False, description='build nghttp2 library (requires C++11)')
-    variant('libssh2', default=False, description='enable libssh2 support')
+    variant('nghttp2',    default=False, description='build nghttp2 library (requires C++11)')
+    variant('libssh2',    default=False, description='enable libssh2 support')
+    variant('libssh',     default=False, description='enable libssh support') #, when='7.58:')
```

Comment:
how can we restrict variants to certain versions? I worked around it by adding the conflict below, but that seems rather ugly...